control/xferfcn.py

```diff
@@ -468,6 +468,82 @@ def __str__(self, var=None):
 
         return outstr
 
+    def to_zpk(self, var=None):
+        """Return string representation of the transfer function as factorized
+        polynomials.
+
+        Examples
+        --------
+        >>> from control import tf
+        >>> G = tf([1],[1, -2, 1])
+        >>> G.to_zpk()
+               1
+        ---------------
+        (s - 1) (s - 1)
+
+        """
+
+        mimo = self.ninputs > 1 or self.noutputs > 1
+        if var is None:
+            # TODO: replace with standard calls to lti functions
+            var = 's' if self.dt is None or self.dt == 0 else 'z'
+        outstr = ""
+
+        for i in range(self.ninputs):
+            for j in range(self.noutputs):
+                if mimo:
```

Comment:
I think where in the library we use zero-based indexing so we should probably do that here.

Reply:
(To clarify: this refers to the next line where printing occurs)

Reply:
I copied the behaviour from how TransferFunction prints mimo systems. 
See https://github.com/python-control/python-control/blob/8972e8292aa6705907a43e00fb5f9c9fc9c259e4/control/xferfcn.py#L443

Better be consistent. Opinion?

Reply:
I see, good point. Math uses 1-based indexing and python uses 0-based. Luckily there is not much ambiguity here because the first one printed out is index 1. 

I guess my take would be to leave it as it is in that case, like it is elsewhere.